cpp/src/rolling/rolling.cu

```diff
@@ -649,12 +998,43 @@ std::unique_ptr<column> rolling_window(column_view const& input,
                                        size_type min_periods,
                                        std::unique_ptr<aggregation> const& agg,
                                        rmm::mr::device_memory_resource* mr)
+{
+  return rolling_window(
+    input, empty_like(input)->view(), preceding_window, following_window, min_periods, agg, mr);
+}
+
+// Applies a fixed-size rolling window function to the values in a column.
+std::unique_ptr<column> rolling_window(column_view const& input,
+                                       column_view const& default_outputs,
+                                       size_type preceding_window,
+                                       size_type following_window,
+                                       size_type min_periods,
+                                       std::unique_ptr<aggregation> const& agg,
+                                       rmm::mr::device_memory_resource* mr)
 {
   CUDF_FUNC_RANGE();
 
   if (input.size() == 0) return empty_like(input);
   CUDF_EXPECTS((min_periods >= 0), "min_periods must be non-negative");
 
+  CUDF_EXPECTS((default_outputs.is_empty() || default_outputs.size() == input.size()),
+               "Defaults column must be either empty or have as many rows as the input column.");
+
+  CUDF_EXPECTS(
+    default_outputs.is_empty() || agg->kind == aggregation::LEAD || agg->kind == aggregation::LAG,
+    "Only LEAD/LAG window functions support default values.");
```

Comment:
Seems like if this parameter is going to be added, all of the aggs should support it.

Reply:
From my limited experience (and based on the specs I'm familiar with), the default value for a `SUM`/`COUNT`/`MIN`/`MAX` isn't semantically defined in SQL.
`LEAD()`/`LAG()` functions are the only ones I'm aware of that allow for (and operate on) default values. E.g. 

- [SQLite](https://sqlite.org/windowfunctions.html)
- Oracle [MIN/MAX](https://oracle-base.com/articles/misc/min-and-max-analytic-functions) vs [LEAD/LAG](https://oracle-base.com/articles/misc/rank-dense-rank-first-last-analytic-functions)

Reply:
With your permission, I'll close this as "resolved". Please feel free to reopen this for discussion. I'd be happy to clarify.